pandas/core/ops.py

```diff
@@ -597,7 +597,15 @@ def _align_method_SERIES(left, right, align_asobject=False):
 
 
 def _construct_result(left, result, index, name, dtype):
-    return left._constructor(result, index=index, name=name, dtype=dtype)
+    """
+    If the raw op result has a non-None name (e.g. it is an Index object) and
+    the name argument is None, then passing name to the constructor will
+    not be enough; we still need to override the name attribute.
+    """
+    out = left._constructor(result, index=index, name=name, dtype=dtype)
```

Comment:
let's just not pass the name at all to the constructor then (and just set as you are doing)

Reply:
Sounds good.